### PR TITLE
Add "--" exec cli support for command arguments

### PR DESCRIPTION
**Pull request by TristanCacqueray**
This flag allows using parameters starting with a - such as:
  ps --args -afx

### CODE REVIEW

**Comment by cyphar** [comment]
I'd prefer if we fix the flag library we're using to support standard Unix `--` semantics, so usage of the library would allow (by default):

```
% runc exec test -- ls -la
```

**Comment by TristanCacqueray** [comment]
I agree this is sub-optimal compared to "--" support. Though this PR enables use-cases when you need to exec command with parameters prefixed by "-".

**Comment by crosbymichael** [comment]
@TristanCacqueray can you add the same logic that is happening here https://github.com/opencontainers/runc/pull/933 to support this without a `--args` flag?

**Comment by TristanCacqueray** [comment]
@crosbymichael sure thing, it's looking better now. Thank you for the review.

**Comment by mrunalp** [comment]
I agree with @cyphar that we should implement standard unix semantics for `--` or switch to another library that supports it.

**Comment by TristanCacqueray** [comment]
@mrunalp it seems like the actual cli library do support "--" and doesn't interpret dashed argument placed after. However it's up to exec.go to remove the extra Args from the list, see 619a523

**Comment by crosbymichael** [comment]
LGTM

@mrunalp it does support it, we just have to get the subslice because we use arg 1 as the container id

[Approved with PullApprove](https://pullapprove.com/opencontainers/runc/pull-request/906/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=opencontainers/runc)

File: man/runc-exec.8.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
    runc exec - execute new process inside the container
 
 # SYNOPSIS
-   runc exec [command options] <container-id> <container command>
+   runc exec [command options] <container-id> <container command> [-- command options]
```
Comment (dqminh) [inline man/runc-exec.8.md]: this usage is  abit weird for `--`. I would expect `runc exec <container-id> -- command options

**Comment by mrunalp** [comment]
To be consistent with other Linux utilities, I think it should support this:
```
[root@dhcp-16-129 ~]# runc exec 2345 -- ls -la
exec failed: exec: "--": executable file not found in $PATH
```
But instead it supports this:
```
[root@dhcp-16-129 ~]# runc exec 2345  ls -- -la
total 124
drwxrwxr-x   21 1000     1000          4096 Aug 25 16:43 .
drwxrwxr-x   21 1000     1000          4096 Aug 25 16:43 ..
-rw-------    1 root     root           156 Jun 10 22:28 .ash_history
drwxrwxr-x    2 1000     1000          4096 Feb 26  2015 bin
drwxr-xr-x    2 1000     1000          4096 Apr 19 17:02 data
drwxr-xr-x    5 root     root           360 Aug 25 16:43 dev
drwxr-xr-x    6 1000     1000          4096 Jul 12 21:57 etc
drwxr-xr-x    2 root     root          4096 Apr 14 18:33 foo
drwxrwxr-x    3 1000     1000          4096 Dec  1  2014 home
drwxrwxr-x    2 1000     1000          4096 Feb 26  2015 lib
lrwxrwxrwx    1 1000     1000             3 Feb 26  2015 lib64 -> lib
lrwxrwxrwx    1 1000     1000            11 Feb 26  2015 linuxrc -> bin/busybox
drwxrwxr-x    2 1000     1000          4096 Dec  1  2014 media
drwxrwxr-x    2 1000     1000          4096 Dec  1  2014 mnt
drwxr-xr-x    2 root     root          4096 Jun 24 23:18 mydata
-rw-------    1 root     root             0 Aug  5 17:17 nohup.out
drwxrwxr-x    2 1000     1000          4096 Dec  1  2014 opt
dr-xr-xr-x  371 root     root             0 Aug 25 16:43 proc
drwx------    2 1000     1000          4096 Mar 18 20:01 root
lrwxrwxrwx    1 1000     1000             3 Dec  1  2014 run -> tmp
drwxr-xr-x    2 1000     1000          4096 Feb 26  2015 sbin
drwxr-xr-x    2 root     root          4096 Jul 12 21:49 srcdata
-rw-r--r--    1 root     root         47858 May 13 19:43 strace.log
dr-xr-xr-x   13 root     root             0 Aug 22 17:08 sys
drwxrwxrwt    3 1000     1000          4096 Feb 26  2015 tmp
drwxrwxr-x    6 1000     1000          4096 Feb 26  2015 usr
drwxrwxr-x    4 1000     1000          4096 Feb 26  2015 var
```

File: man/runc-exec.8.md (hunk 1 above)
Comment (crosbymichael) [inline man/runc-exec.8.md]: I think this help string is just messed up, it should work the way you described.  @TristanCacqueray can you fix the usage strings?

Comment (dqminh) [inline man/runc-exec.8.md]: no it doesnt because `p.Args = append(p.Args[:1], p.Args[2:]...)`. I think @mrunalp also pasted an example :D

Comment (crosbymichael) [inline man/runc-exec.8.md]: ops, i missed that

**Comment by TristanCacqueray** [comment]
@crosbymichael @mrunalp makes sense, I misread the logic of #933...
Thank you for the review, the last patch now support "runc exec 2345 -- ls -la" instead.
Note that the patch doesn't enforce usage of the "--" separator and it checks for its presence before shifting the process args.

**Comment by crosbymichael** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**